Obsidian/Entities/Player.cs

```diff
@@ -304,5 +304,28 @@ public async Task SetGamemodeAsync(Gamemode gamemode)
             await client.QueuePacketAsync(new Net.Packets.Play.Client.GameState.ChangeGamemodeState(gamemode));
             this.Gamemode = gamemode;
         }
+
+
+        public async Task<bool> GrantPermission(string permission)
+        {
+            permission = permission.Trim();
+            bool result = PlayerPermissions.Add(permission);
+            if (result)
+                await this.client.Server.Events.InvokePermissionGrantedAsync(new PermissionGrantedEventArgs(this, permission));
+            return result;
+        }
+        public async Task<bool> RevokePermission(string permission)
+        {
+            permission = permission.Trim();
+            bool result = PlayerPermissions.TryRemove(permission);
+            if (result)
+                await this.client.Server.Events.InvokePermissionRevokedAsync(new PermissionRevokedEventArgs(this, permission));
+            return result;
+        }
+        public Task<bool> HasPermission(string permission)
```

Comment:
this should behave more like namespaces (eg obsidian.ban and obsidian.kick would allow for kick and ban, but obsidian.* would allow for anything under obsidian namespace). Will merge for now and implement that change myself.